src/Database/LSMTree.hs

```diff
@@ -0,0 +1,551 @@
+  , Common.NumEntries (..)
+  , Common.BloomFilterAlloc (..)
+  , Common.defaultBloomFilterAlloc
+  , Common.FencePointerIndex (..)
+  , Common.DiskCachePolicy (..)
+  , Common.MergeSchedule (..)
+  , Common.defaultMergeSchedule
+  , withTable
+  , new
+  , close
+
+    -- * Table queries and updates
+    -- ** Queries
+  , lookups
+  , LookupResult (..)
+  , rangeLookup
+  , Range (..)
+  , QueryResult (..)
+    -- ** Cursor
+  , Cursor
+  , withCursor
+  , withCursorAtOffset
+  , newCursor
+  , newCursorAtOffset
+  , closeCursor
+  , readCursor
+    -- ** Updates
+  , inserts
+  , deletes
+  , mupserts
+  , updates
+  , Update (..)
+    -- ** Blobs
+  , BlobRef
+  , retrieveBlobs
+
+    -- * Durability (snapshots)
+  , SnapshotName
+  , Common.mkSnapshotName
+  , Common.Labellable (..)
+  , createSnapshot
+  , openSnapshot
+  , Common.TableConfigOverride
+  , Common.configNoOverride
+  , Common.configOverrideDiskCachePolicy
+  , deleteSnapshot
+  , listSnapshots
+
+    -- * Persistence
+  , duplicate
+
+    -- * Table union
+  , union
+
+    -- * Serialisation
+  , SerialiseKey
+  , SerialiseValue
+
+    -- * Monoidal value resolution
+  , ResolveValue (..)
+  , resolveDeserialised
+    -- ** Properties
+  , resolveValueValidOutput
+  , resolveValueAssociativity
+    -- ** DerivingVia wrappers
+  , ResolveAsFirst (..)
+
+    -- * Utility types
+  , IOLike
+  ) where
+
+import           Control.DeepSeq
+import           Control.Exception (throw)
+import           Control.Monad
+import           Data.Bifunctor (Bifunctor (..))
+import           Data.Coerce (coerce)
+import           Data.Kind (Type)
+import           Data.Typeable (Proxy (..), eqT, type (:~:) (Refl))
+import qualified Data.Vector as V
+import           Database.LSMTree.Common (BlobRef (BlobRef), IOLike, Range (..),
+                     SerialiseKey, SerialiseValue, Session, SnapshotName,
+                     closeSession, deleteSnapshot, listSnapshots, openSession,
+                     withSession)
+import qualified Database.LSMTree.Common as Common
+import qualified Database.LSMTree.Internal as Internal
+import qualified Database.LSMTree.Internal.BlobRef as Internal
+import qualified Database.LSMTree.Internal.Entry as Entry
+import qualified Database.LSMTree.Internal.RawBytes as RB
+import qualified Database.LSMTree.Internal.Serialise as Internal
+import qualified Database.LSMTree.Internal.Snapshot as Internal
+import qualified Database.LSMTree.Internal.Vector as V
+import           Database.LSMTree.Monoidal (ResolveValue (..),
+                     resolveDeserialised, resolveValueAssociativity,
+                     resolveValueValidOutput)
+
+{-------------------------------------------------------------------------------
+  Tables
+-------------------------------------------------------------------------------}
+
+type Table = Internal.Table'
+
+{-# SPECIALISE withTable ::
+     Session IO
+  -> Common.TableConfig
+  -> (Table IO k v b -> IO a)
```

Comment:
In the other modules we use `blob` for that type parameter. While I generally think it makes sense to use only `b`, only renaming it here, but not in the 90% identical other modules doesn't seem ideal. For example, I looked at the difference of this module to the `Normal` one, and might again in the future to make sure the modules stay in sync. But now this b/blob difference shows up as noise in the diff.

Reply:
Makes sense. Maybe I'll change it back to `blob` here, and make the renaming change in a separate PR

Reply:
Did the renaming in https://github.com/IntersectMBO/lsm-tree/pull/480